src/builder.rs

```diff
@@ -286,7 +286,22 @@ impl SessionBuilder {
         Ok(f(tempdir))
     }
 
-    fn resolve<'a, 'b>(&'a self, mut destination: &'b str) -> (Cow<'a, Self>, &'b str) {
+    /// [`SessionBuilder`] support for `destination` parsing.
+    /// The format of `destination` is the same as the `destination` argument to `ssh`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use std::borrow::Cow;
+    /// use openssh::SessionBuilder;
+    /// let b = SessionBuilder::default();
+    /// let (b, d) = b.resolve("ssh://test-user@127.0.0.1:2222");
+    /// assert_eq!(d, "127.0.0.1");
+    /// assert_eq!(b, Cow::Borrowed(SessionBuilder::default()
```

Comment:
You could instead, add a getter for retrieving `port` and `user` of `SessioBuilder`.